benchmarks/benchmark_dataset.py

```diff
@@ -1000,6 +1001,142 @@ def sample(
         return sampled_requests
 
 
+# -----------------------------------------------------------------------------
+# Azure Dataset Implementation
+# -----------------------------------------------------------------------------
+
+
+class AzureDataset(BenchmarkDataset):
+    """
+    Azure Dataset.
+    https://github.com/Azure/AzurePublicDataset/blob/master/AzureLLMInferenceDataset2023.md
+
+    Implements the Azure dataset. Loads data from a csv file and generates
+    sample requrests based on the Azure dataset format. E.g.,
+    ```
+    TIMESTAMP,ContextTokens,GeneratedTokens
+    2023-11-16 18:17:03.9799600,4808,10
+    2023-11-16 18:17:04.0319600,3180,8
+    2023-11-16 18:17:04.0781490,110,27
+    ```
+    """
+
+    def __init__(self, **kwargs) -> None:
+        super().__init__(**kwargs)
+        self.load_data()
+
+    def load_data(self) -> None:
+        if self.dataset_path is None:
+            raise ValueError("dataset_path must be provided for loading data.")
+
+        self.data = []
+
+        # Load the CSV file
+        csv_data = pd.read_csv(self.dataset_path)
+        self.interval_data = csv_data[["TIMESTAMP"]].copy()
+        for _, row in csv_data.iterrows():
+            self.data.append(row.to_dict())
+
+        # Shuffle the data for randomness
+        random.seed(self.random_seed)
+        random.shuffle(self.data)
+
+    def _apply_time_filters(
+        self, start_time: Optional[str] = None, end_time: Optional[str] = None
+    ) -> None:
+        """
+        Filter the dataset based on start and end timestamps if needed.
+        """
+        if start_time is None and end_time is None:
+            return
+
+        if start_time is not None:
+            self.data = [item for item in self.data if item["TIMESTAMP"] >= start_time]
+            self.interval_data = self.interval_data[
+                self.interval_data["TIMESTAMP"] >= start_time
+            ]
+        if end_time is not None:
+            self.data = [item for item in self.data if item["TIMESTAMP"] <= end_time]
+            self.interval_data = self.interval_data[
+                self.interval_data["TIMESTAMP"] <= end_time
+            ]
+
+        assert len(self.data) == len(self.interval_data), (
+            "Data and interval data lengths do not match after filtering."
+        )
+
+    def _compute_time_intervals(self) -> list:
+        """
+        Compute the inter-request time based on the TIMESTAMP column.
+        This method modifies the interval_data DataFrame in place.
+        """
+        self.interval_data["TIMESTAMP"] = pd.to_datetime(
+            self.interval_data["TIMESTAMP"]
+        )
+        self.interval_data["time_interval"] = (
+            self.interval_data["TIMESTAMP"].diff().dt.total_seconds().fillna(0)
+        )
+        return self.interval_data["time_interval"].tolist()
```

Comment:
```suggestion
    def _compute_time_intervals(self) -> list:
        """
        Compute the inter-request time based on the TIMESTAMP column.
        This method modifies the interval_data DataFrame in place.
        """
        return (
            pd.to_datetime(self.interval_data["TIMESTAMP"])
            .diff()
            .dt.total_seconds()
            .fillna(0)
            .tolist()
        )
```

`self.interval_data` 를 매번 덮어 쓸 필요가 없는 듯 하여 더 간결하게 줄여보았습니다.